website/docs/developer/development-guide/code/best-practices.mdx

```diff
@@ -0,0 +1,72 @@
+---
+id: best-practices
+title: Best Practices
+description: Best practices for coding in ORCA.
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+## Unit Testing
+All code should reach minimum 80% coverage through [Unit Tests](unit-tests.md).
+
+## Code Style
+We use the [Google Style Guide](https://google.github.io/styleguide/pyguide.html) for style elements such as documentation, titling, and structure.
+
+## Stop on Failure
+Failures within ORCA break through to the Cumulus workflow they are a part of.
+To this end, raising an error is preferred over catching the error and returning a null value or error message.
+
+```python
+try:
+    value = function(param)
+except requests_db.DatabaseError as err:
+    logging.error(err)
+    raise
+```
+```python
+if not success:
+    logging.error(f"You may log additional information if desired. "
+                  f"param: {param}")
+    raise DescriptiveErrorType(f'Error message to be raised info Cumulus workflow.'
+```
+Retries can then be configured in the workflow json if desired. See
+[documentation](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html)
+and
+[tutorials](https://aws.amazon.com/getting-started/hands-on/handle-serverless-application-errors-step-functions-lambda/)
+for more information.
+```json
+"SomeState": {
+  ...
+  "Retry": [
+    {
+      "ErrorEquals": [
+        "Lambda.DescriptiveErrorType",
+        "Lambda.AnotherErrorType"
+      ],
+      "IntervalSeconds": 2,
+      "MaxAttempts": 6,
+      "BackoffRate": 2
+    }
+  ],
+  "Catch": [
+    {
+      "ErrorEquals": [
+        "States.ALL"
+      ],
+      "ResultPath": "$.exception",
+      "Next": "WorkflowFailed"
+    }
+  ],
+  "Next": "NextState"
+}
+```
+If the retries are exceeded and the error is caught, then the workflow will show that it jumped to the 'WorkflowFailed' state.
```

Comment:
Change  'WorkflowFailed' to `WorkflowFailed` so it comes across as code.

Reply:
Done